src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -291,8 +294,11 @@
 (defn post-req->query
   "Takes a POST body and parses the JSON to create a pdb query map"
   [req parse-fn]
-  (-> (with-open [reader (-> req :body clojure.java.io/reader)]
-        (json/parse-stream reader true))
+  (-> (let [req-body (request/body-string req)]
+        (try
+          (json/parse-string req-body true)
+        (catch JsonParseException e
+          (throw (IllegalArgumentException. (tru (pprint-json-parse-exception e req-body)))))))
```

Comment:
Wrapping here won't identify the string for translation properly and thus won't be translated.  The extraction process expects strings to be embedded directly in the `tru` `trs`, etc functions.

Reply:
ah, yep